yarn-project/prover-client/src/orchestrator/orchestrator_errors.test.ts

```diff
@@ -1,144 +1,128 @@
+import { makeBloatedProcessedTx, makeEmptyProcessedTestTx } from '../mocks/fixtures.js';
+import { TestContext } from '../mocks/test_context.js';
 
 export const createMemDown = () => (memdown as any)() as MemDown<any, any>;
 
-const logger = createDebugLogger('aztec:orchestrator-test');
+const logger = createDebugLogger('aztec:orchestrator-errors');
 
-describe('prover/orchestrator', () => {
-  let builder: ProvingOrchestrator;
-  let builderDb: MerkleTreeOperations;
-
-  let prover: TestCircuitProver;
-
-  let blockNumber: number;
-
-  let globalVariables: GlobalVariables;
+describe('prover/orchestrator/errors', () => {
+  let context: TestContext;
 
   beforeEach(async () => {
-    blockNumber = 3;
-    globalVariables = makeGlobals(blockNumber);
-
-    const acvmConfig = await getConfig(logger);
-    const simulationProvider = await getSimulationProvider({
-      acvmWorkingDirectory: acvmConfig?.acvmWorkingDirectory,
-      acvmBinaryPath: acvmConfig?.expectedAcvmPath,
-    });
-    prover = new TestCircuitProver(simulationProvider);
-
-    builderDb = await MerkleTrees.new(openTmpStore()).then(t => t.asLatest());
-    builder = new ProvingOrchestrator(builderDb, prover, 1);
+    context = await TestContext.new(logger);
   }, 20_000);
 
-  describe('errors', () => {
-    beforeEach(async () => {
-      builder = await ProvingOrchestrator.new(builderDb, prover);
-    });
+  afterEach(async () => {
+    await context.cleanup();
+  });
 
-    afterEach(async () => {
-      await builder.stop();
-    });
+  afterAll(async () => {});
 
+  describe('errors', () => {
     it('throws if adding too many transactions', async () => {
       const txs = await Promise.all([
-        makeBloatedProcessedTx(builderDb, 1),
-        makeBloatedProcessedTx(builderDb, 2),
-        makeBloatedProcessedTx(builderDb, 3),
-        makeBloatedProcessedTx(builderDb, 4),
+        makeBloatedProcessedTx(context.actualDb, 1),
+        makeBloatedProcessedTx(context.actualDb, 2),
+        makeBloatedProcessedTx(context.actualDb, 3),
+        makeBloatedProcessedTx(context.actualDb, 4),
       ]);
 
-      const blockTicket = await builder.startNewBlock(
+      const blockTicket = await context.orchestrator.startNewBlock(
         txs.length,
-        globalVariables,
+        context.globalVariables,
         [],
-        await makeEmptyProcessedTestTx(builderDb),
+        await makeEmptyProcessedTestTx(context.actualDb),
       );
 
       for (const tx of txs) {
-        await builder.addNewTx(tx);
+        await context.orchestrator.addNewTx(tx);
       }
 
-      await expect(async () => await builder.addNewTx(await makeEmptyProcessedTestTx(builderDb))).rejects.toThrow(
-        'Rollup not accepting further transactions',
-      );
+      await expect(
+        async () => await context.orchestrator.addNewTx(await makeEmptyProcessedTestTx(context.actualDb)),
+      ).rejects.toThrow('Rollup not accepting further transactions');
```

Comment:
TIL `expect().rejects` calls a function if given one. I always thought `resolves` and `rejects` only work on promises.